### DISCUSSION TITLE
Support for AMD Radeon RX 7800M GPU

### DISCUSSION THREAD

**Discussion by ardimaunahan** [ROCm/ROCm#4071] [Q&A]
Hi. I recently got my OneXGPU 2 which is using AMD Radeon RX 7800M GPU. I tested ROCm 6.2.4 in my Win 11 WSL instance and I'm getting this error:

> $ rocminfo
> WSL environment detected.
> hsa api call failure at: /long_pathname_so_that_rpms_can_package_the_debug_info/src/rocminfo/rocminfo.cc:1306
> Call returned HSA_STATUS_ERROR_OUT_OF_RESOURCES: The runtime failed to allocate the necessary resources. This error may also occur when the core runtime library needs to spawn threads or create internal OS-specific events.

Any tips on how to make it work? Should I try in an actual Ubuntu instead of WSL? I'm willing to be a tester if someone is working to support my hardware.

Note that when I use LM Studio 0.3.5, it seems it can leverage my GPU somehow.

**Comment by chicheng** [comment]
Try ROCm 6.1.3 for WSL: https://rocm.docs.amd.com/projects/radeon/en/latest/docs/compatibility/wsl/wsl_compatibility.html

**Reply by schung-amd** [reply]
As @chicheng notes, WSL support is very specific, and currently the only ROCm version that works with WSL is 6.1.3. In addition, you will need to install Adrenalin 24.6.1 (or its preview counterpart in the compatibility matrix, Adrenalin Edition 24.10.21.01 for WSL 2) as stated in https://rocm.docs.amd.com/projects/radeon/en/latest/docs/install/wsl/install-radeon.html. The 7800M is also not officially supported (i.e. we don't test for this GPU), but this may work anyway.

**Comment by ardimaunahan** [comment]
Thanks guys for the reply. I just tried the latest release of AMD Adrenalin Driver now, but it seems, 7800M is still not supported... 

![ScreenHunter_811 Dec  06 19 49](https://github.com/user-attachments/assets/76237966-52f0-4954-b32a-ea910119d59c)
![ScreenHunter_810 Dec  06 19 49](https://github.com/user-attachments/assets/c0e3a698-7f3c-47fd-881e-9cd69db7089d)
![ScreenHunter_809 Dec  06 19 48](https://github.com/user-attachments/assets/73606577-88b3-41e3-ba4f-5718dc178998)

I wish it will be supported in the future. Or maybe I missed some other steps in either inside WSL or in Windows 11? Let me know if there's some other ways to troubleshoot my setup.

Or if there's anyone who has the setup mentioned in this official list, please let me know if you can successfully use ROCm in WSL 2. Thanks!
![ScreenHunter_812 Dec  06 19 55](https://github.com/user-attachments/assets/eb8a8700-ded2-4590-81eb-e6559e087c99)

**Reply by schung-amd** [reply]
I haven't tried the new Adrenalin driver myself, but 6.2.3 does work with the 24.6.1 driver on supported hardware. Unfortunately, the 7800M is a different architecture and not currently supported. We're aware that the current hardware requirements are quite restrictive and are looking to expand this list in the future, and hopefully this extends to the 7800M.

**Reply by ardimaunahan** [reply]
Thanks, looking forward to it. I guess for now, this is just for gaming and LM Studio. But I wonder how LM Studio able to support my GPU via this "runtime":

![ScreenHunter_813 Dec  06 22 44](https://github.com/user-attachments/assets/884081c0-03dd-4766-8375-f063a0cdaddd)

**Reply by schung-amd** [reply]
Is that in WSL or natively in Windows? We have much broader hardware support for HIP SDK in Windows: https://rocm.docs.amd.com/projects/install-on-windows/en/develop/reference/system-requirements.html. Despite the naming, this provides a subset of ROCm packages as well.

**Reply by ardimaunahan** [reply]
Natively in Windows 11. Oh, that's good then. I'll try to learn how to use the packages in Windows then. 

My planned use case is for ML training using either TensorFlow or PyTorch. Also, it would be great if ollama can also leverage it.

**Reply by schung-amd** [reply]
Not sure about tensorflow, but pytorch is not natively supported on Windows through ROCm yet. ollama uses llama.cpp as a backend afaik, so it should work.

**Reply by brianangulo** [reply]
Hey @ardimaunahan I'm running into the same issues so ended up dual booting ubuntu alongside and that works great with rocm for pytorch. Just in case you want to try it.

**Reply by brianangulo** [reply]
@schung-amd just curious is there a way to maybe via compiling from source that we can enable rocm support for 7800 cards through wsl2? Just like we can use HSA_OVERRIDE flag on linux? Thanks

**Reply by schung-amd** [reply]
@brianangulo I'm not aware of any similar workarounds at the moment, sorry.

**Comment by harkgill-amd** [comment]
We've made large strides in both GPU and WSL support since this discussion was first opened. Please give the latest release below a try (same steps as Linux for WSL) and feel free to open a GitHub issue if you run into any difficulties.

https://rocm.docs.amd.com/en/latest/install/rocm.html
